flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -254,22 +259,33 @@ public void submitSQLJob(SQLJobSubmission job) throws IOException {
 		AutoClosableProcess.runBlocking(commands.toArray(new String[0]));
 	}
 
-	public void copyOptJarsToLib(String jarNamePrefix) throws FileNotFoundException, IOException {
-		final Optional<Path> reporterJarOptional;
+	public void copyOptJarsToLib(String jarNamePrefix) throws IOException {
+		final Path jar = getOptJar(jarNamePrefix);
+		Files.copy(jar, lib.resolve(jar.getFileName()));
+	}
+
+	public void addPlugin(String jarNamePrefix) throws IOException {
+		final Path jar = getOptJar(jarNamePrefix);
+		final Path pluginPath = plugins.resolve(jarNamePrefix);
+		Files.createDirectory(pluginPath);
+		Files.copy(jar, pluginPath.resolve(jar.getFileName()));
+	}
+
+	@Nonnull
+	public Path getOptJar(final String jarNamePrefix) throws IOException {
```

Comment:
you should never expose file paths; people usually do stupid things when they have access to that.